### PR TITLE
lib.sh: disable logging with SOF_LOGGING=none env or missing .ldc

### PR DESCRIPTION
**Pull request by marc-hb**
Some configurations don't support the logger. Others do but do not want
to run it for various reasons.

Supersedes #811, see earlier discussions there.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
Need to fix the commit message.

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR813/build934/devicetest has two known issues: ADL NTP sync / too fast boot + very recent (5.6) "preemptible" regression https://github.com/thesofproject/linux/issues/3283

To really test this in CI (in addition to extensive testing locally) I hardcoded `SOF_LOGGING=none` in throw-away PR #814. The results in https://sof-ci.01.org/softestpr/PR814/build935/devicetest/ look good: all tests passed or skipped without any FW logs - except for the (very special/different) sof-logger test which passed with FW logs.

File: case-lib/lib.sh
Comment (greg-intel) [inline case-lib/lib.sh:406]: Not sure this comment is needed, as the script self documents on the line before.

**Review by greg-intel — APPROVED** [review]

File: case-lib/lib.sh
Comment (fredoh9) [inline case-lib/lib.sh:415]: Need more clarification. In future we will support only etrace? why? slogger seems useful to me

**Review by fredoh9 — APPROVED** [review]
In general, I like this idea. logger_disabled() has 3 options to support
1. no ldc file in the system
2. OPT_VAL['s'] == 0
3. $SOF_LOGGING == none

File: case-lib/lib.sh
Comment (marc-hb) [reply case-lib/lib.sh:415]: See #726 referenced on the next line.

File: case-lib/lib.sh
Comment (marc-hb) [reply case-lib/lib.sh:406]: You're right, it's overkill. I will remove but only if someone requests some other changes because it does do not much harm either and uses very little real estate.

**Review by marc-hb — COMMENTED** [review]
Thanks for the quick reviews. As this (and #811 before it) is eagerly waited, I will merge in 6-7 hours if no one requests any change. Sorry for the short notice (not so short when including earlier discussions in #811)

**Review by aiChaoSONG — APPROVED** [review]

**Pull request merged**
